compiler/record-minmax/include/MinMaxComputer.h

```diff
@@ -33,8 +38,8 @@ class MinMaxComputer
   virtual ~MinMaxComputer() = default;
 
   // Child class must implement this
-  // TODO Use proper input signature
```

Comment:
This PR resolves this TODO.

`MinMaxObserver` and `MinMaxComputer` are not coupled anymore, which makes it easy to write unittests for MinMaxComputer.